src/time_zone_info.cc

```diff
@@ -752,70 +749,54 @@ class FuchsiaZoneInfoSource : public FileZoneInfoSource {
   std::string Version() const override { return version_; }
 
  private:
-  explicit FuchsiaZoneInfoSource(FILE* fp, const char* vers)
-      : FileZoneInfoSource(fp), version_(vers) {}
-
+  explicit FuchsiaZoneInfoSource(FilePtr fp, std::string version)
+      : FileZoneInfoSource(std::move(fp)), version_(std::move(version)) {}
   std::string version_;
 };
 
-// Reads the first line of the file into a string. Returns an empty string on
-// error.
-std::string ReadFirstLine(const std::string& path) {
-  std::string line;
-  std::ifstream input_stream(path);
-  if (input_stream.good()) {
-    std::getline(input_stream, line);
-  }
-  return line;
-}
-
 std::unique_ptr<ZoneInfoSource> FuchsiaZoneInfoSource::Open(
     const std::string& name) {
-  // Locations where a Fuchsia component might find zoneinfo files, in
-  // descending order of preference.
-  const auto kTzdataPathPrefixes = {
+  // Use of the "file:" prefix is intended for testing purposes only.
+  const std::size_t pos = (name.compare(0, 5, "file:") == 0) ? 5 : 0;
+
+  // Prefixes where a Fuchsia component might find zoneinfo files,
+  // in descending order of preference.
+  const auto kTzdataPrefixes = {
       "/config/data/tzdata/",
       "/pkg/data/tzdata/",
       "/data/tzdata/",
   };
-  // Used for tests.
-  const auto kEmptyPathPrefixes = {""};
-  // Location under <tzdata> where tzif2 files are stored.
-  const auto kZoneinfoFormatPrefix = "zoneinfo/tzif2/";
-  // Fuchsia builds place the version string in <tzdata>/revision.txt.
-  const auto kVersionFileName = "revision.txt";
-
-  // Use of the "file:" prefix is intended for testing purposes only.
-  const std::size_t pos = (name.compare(0, 5, "file:") == 0) ? 5 : 0;
-  const bool is_absolute_path = (pos != name.size() && name[pos] == '/');
-  const auto path_prefixes =
-      is_absolute_path ? kEmptyPathPrefixes : kTzdataPathPrefixes;
-
-  for (const char* tzdata_dir : path_prefixes) {
-    // Map the time-zone name to a path name.
-    // Fuchsia builds place time zone files at <tzdata>/<format>/...
-    std::string path = tzdata_dir;
-    if (!is_absolute_path) {
-      path += kZoneinfoFormatPrefix;
-    }
+  const auto kEmptyPrefix = {""};
+  const bool name_absolute = (pos != name.size() && name[pos] == '/');
+  const auto prefixes = name_absolute ? kEmptyPrefix : kTzdataPrefixes;
+
+  // Fuchsia builds place zoneinfo files at "<prefix><format><name>".
+  for (const std::string prefix : prefixes) {
```

Comment:
`prefix` could be a const-ref here. I don't think it needs to be a value.

Reply:
I don't think it makes a difference either way, as in each case we're just constructing a `std::string` from a `const char*`.  I usually prefer to give that constructed value a name rather than binding a reference to a temporary, but OK.